config-model/src/main/java/com/yahoo/vespa/model/container/component/BindingPattern.java

```diff
@@ -0,0 +1,127 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.model.container.component;
+
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * URI binding pattern used by filter and handler bindings.
+ *
+ * There are two types of binding; user generated and model generated bindings.
+ * - User generated bindings are bindings which are constructed from directly from 'binding' elements from services.xml
+ * - Model generated bindings are binding which are implicitly constructed by the model, e.g built-in handlers.
+ *
+ * @author bjorncs
+ */
+public class BindingPattern implements Comparable<BindingPattern> {
+
+    private static final Pattern BINDING_PATTERN =
+            Pattern.compile("([^:]+)://([^:/]+)(:((\\*)|([0-9]+)))?(/.*)", Pattern.UNICODE_CASE | Pattern.CANON_EQ);
+
+    private final String scheme;
+    private final String host;
+    private final String port;
+    private final String path;
+    private final boolean isUserGenerated;
+
+    private BindingPattern(
+            String scheme,
+            String host,
+            String port,
+            String path,
+            boolean isUserGenerated) {
+        this.scheme = Objects.requireNonNull(scheme, "Scheme in binding must be specified");
+        this.host = Objects.requireNonNull(host, "Host must be specified");
+        this.port = port;
+        this.path = validatePath(path);
+        this.isUserGenerated = isUserGenerated;
+    }
+
+    private static String validatePath(String path) {
+        Objects.requireNonNull(path, "Path must be specified");
+        if (!path.startsWith("/")) throw new IllegalArgumentException("Path has not '/' as prefix: " + path);
+        return path;
+    }
+
+    public static BindingPattern createUserGeneratedFromPattern(String pattern) {
+        return createFromBindingString(pattern, true);
+    }
+
+    public static BindingPattern createUserGeneratedFromHttpPath(String path) {
+        return new BindingPattern("http", "*", null, path, true);
+    }
+
+    public static BindingPattern createModelGeneratedFromPattern(String pattern) {
+        return createFromBindingString(pattern, false);
+    }
+
+    public static BindingPattern createModelGeneratedFromHttpPath(String path) {
+        return new BindingPattern("http", "*", null, path, false);
+    }
```

Comment:
These method names are rather clunky. Perhaps it would be better to have just two methods that take the type as an extra argument?
```
fromPattern(String pattern, Type type)
fromHttpPath(String path, Type type)
```